### PR TITLE
I2C Fix

### PR DESCRIPTION
**Pull request by CrasherCourse**
I believe that these are all the changes we needed from hacky.
Please notify me if there is something amiss.

### CODE REVIEW

File: HeadMaster.c [hunk 1]
```diff
@@ -146,7 +146,7 @@ void main(void)
         DMXReceive(&dmxDevice);
         HMMapDmxToServo(dmxDevice.buffer, dmxDevice.bufferSize);
         for (receiver = 0; receiver < NUMBER_OF_SLAVES; ++receiver) {
-            I2CSend(receiver);
+            I2CSend(receiver, dmxDevice.buffer);
```
Comment (kevana) [inline HeadMaster.c]: Being pedantic, but I think I remember @tylrtrmbl wanting the buffer being the first argument of the function. Also, passing by value vs. reference. We're not using interrupts so data consistency isn't a problem, but won't copying the entire array give a performance hit?

**Comment by taylortrimble** [comment]
I think we could pick a better name for `waitForSPIF`. To match coding conventions, how about `I2CWaitForTransmission` or something? Also, I kind of would like the buffer first.

**Comment by taylortrimble** [comment]
Also, I don't think we'll ever have a situation where we send a byte without waiting for the transmission to complete. So let's move the waiting after byte transmissions to inside `I2CSendByte`.

**Comment by taylortrimble** [comment]
For the rest of style, can we move the waiting function to the bottom since it's not that interesting, forward declare each of the functions after the `Code` banner, and fix the curly brace on `I2CSendByte`'s implementation?

**Comment by taylortrimble** [comment]
Functionally: let's convert to using the slave's number as it's address. It will prevent a **lot** of pointer swapping in the PIC. After we do that, both of the global variables in `I2C.h` are invalid.

**Comment by CrasherCourse** [comment]
Actually I named it `waitForSPIF` because sending the start and stop condition will trigger the interrupt.
Meaning the function is used for more than waiting for a transmission.
Maybe someone has a better name for this or I just copy and paste the code to replace the function itself.

**Comment by taylortrimble** [comment]
Technically it's "transmitting" START and STOP conditions as well, so the "wait for transmission" language works all right. Other suggestions welcome too.

Let's forward declare functions at the top of the code block and merge our wait function into sending a byte now (discussed above)

Thanks @CrasherCourse!

**Comment by CrasherCourse** [comment]
I actually put the waiting function into the header.  Instead of declaring it at the top of the code block.

I still have the wait function after the start and stop transmissions since the don't call `I2CsendByte`.

**Comment by taylortrimble** [comment]
It all looks good! Our coding conventions say to re-declare all the functions again after the code banner, which is dumb. So you can leave them out and I'll change the coding conventions.

If you tidy up the `I2C.h` bottom to look more like the `Defines` section and remove the now-useless `Global Variables` banner, this should be neat enough to merge.

**Comment by CrasherCourse** [comment]
Merging soon.  Speak now or forever hold your peace.

**Comment by taylortrimble** [comment]
Okay, I fixed the things I still wanted changed:
- Newline at end of file
- No banner close thingy
- Closer function spacing

**PLUS** one important change: `I2CSendByte` and `I2CWaitForTransmission` aren't really _user_ functions. As such, they aren't part of an API per se and don't belong in the header file— which was the original rationale behind placing the function declarations at the top of the code banner. Does this make sense?

Just remembered that old reasoning now!

**Comment by CrasherCourse** [comment]
@tylrtrmbl Makes perfect sense.

**Comment by CrasherCourse** [comment]
But now I got this `DETACHED HEAD:` branch thing my Git Hub application and I can't go back to the `I2Cfix` branch.

Also, the curly braces of `I2CSend` should be aligned.  I can't fix it at the moment do to stupid stuff.

Edit: Never mind just fixed that issue.

**Comment by CrasherCourse** [comment]
Now I can't sync for some reason (behind a commit?)

**Comment by CrasherCourse** [comment]
@tylrtrmbl you might want to put the finishing touches for me.

My local branch has become buggy and won't sync.

**Comment by taylortrimble** [comment]
try `git pull origin I2Cfix` at the command line

**Comment by CrasherCourse** [comment]
:smile:  It worked!

Also, I accidentally learned how to do that smile icon.

**Pull request merged**